td-shim/src/bin/td-shim/memory.rs

```diff
@@ -286,6 +287,16 @@ impl<'a> Memory<'a> {
         use td_layout::runtime::TD_PAYLOAD_PARTIAL_ACCEPT_MEMORY_SIZE;
         use td_uefi_pi::pi;
 
+        // The physical address must not exceed the shared mask (the last bit of GPAW).
+        let max_phys_addr = resources
+            .iter()
+            .map(|resource| resource.physical_start + resource.resource_length - 1)
+            .max()
+            .unwrap();
+        if max_phys_addr > td::get_shared_page_mask() {
+            panic!("Invalid physical address for TDX");
```

Comment:
- Would it be useful to log `max_phys_addr` in the panic call?
- If so, would it be useful to log the resource index / indices of the invalid resources too?